### ISSUE TITLE
Genotype likelihod error

### ISSUE THREAD

**Issue by mcetincompbio** [ANGSD/angsd#176]
Hi,

I am running the below command to calculate genotype likelihoods from a bam file using SAMtools model.

./angsd -GL 1 -bam /mnt/NAS/share/DNA/WGS/BAM/ancient_hsa/Kilinc2016/Bon005_031102_15.merged.hs37d5.fa.cons.90perc.bam
 
I get the following error:
[E::hts_open_format] Failed to open file  _(unreadable character here)_
angsd: multiReader.cpp:348: multiReader::multiReader(int, char**): Assertion `in' failed.
Aborted (core dumped)

I would greatly appreciate help.

**Comment by ANGSD** [comment]
-bam assumes a list of bam/cram files. For a single file please use -i
your.bam instead.

Best

2018-08-20 13:24 GMT+02:00 mcetin84 <notifications@github.com>:

**Comment by mcetincompbio** [comment]
Thanks!

It worked.

Best

**Comment by dlema82** [comment]
Hello,
I am having the same problem, except for the fact that I AM using the -bam command with a bam.filelist, but I still get the error "no such file or directory" when it reads the files in my bam.filelist. I've edited the file list with the full/path/to/each/bam, and added \n to each line. I've tried it with and without a .txt extension, I checked and rechecked again and again for any spelling or typo errors for each file and their paths. I tried dos2unix fix on the bam.filelist. Nothing works. Angsd will not read in my bam files from a bam.filelist with the -bam command. What am I doing wrong here??? I cannot find a solution anywhere.

**Comment by Jungal10** [comment]
Not sure if I am still on time, but I had this error when not using absolute paths. Maybe is that?

**Comment by soraiabarbosa** [comment]
I got the same error simply by downsampling my bam.filelist file. I.e. when using a full list of bams with absolute paths (i.e. all the bams I have across multiple folders), it works. But if I simply remove a few lines out of that list, I get an error for whatever the first bam on the list is.

**Comment by ccfitzcook** [comment]
@ANGSD Did anyone get an answer to this issue? I am having the same problem as  @soraiabarbosa

**Comment by dlema82** [comment]
I have found that the program gets funky with file paths. I put all my bams in the same directory as my job submission (working directory), and in my bamlist they are listed without file paths, so that it reads them directly in the folder the job is running in and this works. I think when file paths are too long/have too many folder extensions, it cannot easily read them for whatever reason.

**Comment by ccfitzcook** [comment]
@dlema82 I don't think my issue was with file paths - I had already done what you described before commenting, as well as trying to use absolute paths and relative paths with no luck. 

While troubleshooting, I re-downloaded the newest version of angsd and re-indexed the reference genome using samtools-1.11 instead of samtools 1.10 which is the default on my computing cluster. I think there was an issue with having mismatched angsd/samtools versions. However, this is rather strange since I was able to use angsd on the full bams list before subsetting without issue before making those changes with angsd and samtools versions. 

I also tried subsetting the bams file into separate files using grep commands and piping the outputs to new files rather than what I was doing before, which was manually subsetting them in a text editor (Notepad++). Not sure which of those changes solved the problem, but in the end, things worked!
